### PR TITLE
refactor: rename tests with error assertions

### LINKED ISSUE #155: [Refactor]: apply same naming convention for all tests that check errors
### What is the feature you would like to see?

We should apply the same naming convention for all tests that check errors:
`fn error_when_<function>_<description>(...)`

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines

### PR DESCRIPTION
**Pull request by postmeback**
Changes as per the mentioned issue. It's a work in progress, requesting a quick review. Will continue working on it, if the initial changes are fine.

Resolves #155 

#### PR Checklist

The below items are not applicable as such, because it is more about renaming tests to a particular naming convention.

- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: contracts/src/token/erc721/mod.rs [hunk 1]
```diff
@@ -1141,21 +1141,18 @@ mod tests {
     }
 
     #[motsu::test]
-    fn balance_of_zero_balance(contract: Erc721) {
+    fn error_when_contract_balance_is_zero_balance(contract: Erc721) {
```
Comment (qalisander) [inline contracts/src/token/erc721/mod.rs:1144]: I can see this test is not checking any error. I think we are not expecting to change its name. cc @bidzyyys

File: contracts/src/token/erc721/mod.rs [hunk 2]
```diff
@@ -1182,17 +1178,17 @@ mod tests {
...
     #[motsu::test]
-    fn error_when_minting_token_id_twice(contract: Erc721) {
+    fn error_when_minting_same_token_id_twice(contract: Erc721) {
         let alice = msg::sender();
-        let token_id = random_token_id();
+        let token_id = random_token_id();   
         contract
             ._mint(alice, token_id)
-            .expect("should mint the token a first time");
+            .expect("should mint the token the first time");
```
Comment (qalisander) [inline contracts/src/token/erc721/mod.rs:1191]: good

File: contracts/src/token/erc721/mod.rs [hunk 3]
```diff
@@ -1165,14 +1162,13 @@ mod tests {
     }
 
     #[motsu::test]
-    fn mints(contract: Erc721) {
+    fn when_minting_token_balance_and_owner_are_correct(contract: Erc721) {
```
Comment (qalisander) [inline contracts/src/token/erc721/mod.rs:1165]: In this test we're also not checking for any error. Seems make sense to leave it as it is. cc @bidzyyys

**Review by qalisander — COMMENTED** [review]
Hey @postmeback!
Appreciate your contribution!
Here are few comments I have:

File: contracts/src/token/erc721/mod.rs (hunk 3 above)
Comment (bidzyyys) [reply contracts/src/token/erc721/mod.rs:1165]: `mints` is fine here. The task is to rename tests that checks errors.

File: contracts/src/token/erc721/mod.rs (hunk 1 above)
Comment (bidzyyys) [reply contracts/src/token/erc721/mod.rs:1144]: Yep, we should rename only when we check errors in tests.

**Comment by alexfertel** [comment]
Hey @postmeback! We're really sorry about this, but after some discussion, we found that renaming the tests this way hides away the intent of the test, so we don't think it's worth the trade-off.

As such, I'll be closing the PR, since we won't be proceeding with these changes anymore. Hopefully this doesn't deter you from contributing to our library! Thanks.

**Pull request closed without merging**